picard/file.py

```diff
@@ -225,24 +225,21 @@ def _loading_finished(self, callback, result=None, error=None):
         self.update()
         callback(self)
 
-    def _copy_loaded_metadata(self, metadata):
-        filename, _ = os.path.splitext(self.base_filename)
+    def _copy_loaded_metadata(self, metadata, guess_tracknumber_and_title=True):
```

Comment:
Instead of passing this option, I'd pass a list of functions to run on metadata before copying it. It would be much more generic.

Something like:

```python
def _copy_loaded_metadata(self, metadata):
    metadata['~length'] = format_time(metadata.length)
    for processor in self.copy_loaded_metadata_processors:
          processor(metadata)
```

And in `_loading_finished()`:

```python
self.copy_loaded_metadata_processors = list()
if config.setting["guess_tracknumber_and_title"]:
  self.copy_loaded_metadata_processors.append(self._guess_tracknumber_and_title)
```

Reply:
Good idea, that parameter made me unhappy already. I'll change that.

Reply:
I did it a bit different in https://github.com/metabrainz/picard/pull/1798/commits/a73e65fcc3594923823081769c4ab22faad8b7d6 . I wanted to avoid using an instance variable to implicitly pass around parameters, instead do this explicitly.